### ISSUE TITLE
Server side rendering: Link activeStyle is inherited

### ISSUE THREAD

**Issue by Vanuan** [apollostack/react-apollo#115]
I don't know if it's specific to apollo or the router, but it looks like there's some discrepancy between rendering "Link" on server and in the browser:

> warning.js:44Warning: React attempted to reuse markup in a container but the checksum was invalid. This generally means that you are using server rendering and the markup generated on the server was not what the client was expecting. React injected new markup to compensate which works but you have lost many of the benefits of server rendering. Instead, figure out why the markup being generated is different on the client or server:

Browser:
```javascript
match({ history: browserHistory, routes }, (error, redirectLocation, renderProps) => {
  render((<ApolloProvider store={store} client={client}>
    <Router {...renderProps} />
  </ApolloProvider>), document.getElementById('container'));
})
```

Server:
```javascript
  match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
    if (error) {
      console.error(error);
      res.status(500).send("500 Internal error")
    } else if (redirectLocation) {
      res.redirect(302, redirectLocation.pathname + redirectLocation.search)
    } else if (renderProps) {
      var app = (<ApolloProvider store={store} client={client} {...renderProps}>
        <RouterContext {...renderProps} />
      </ApolloProvider>);
      getDataFromTree(app).then(function({store, client}) {
        let htmlString = ReactDOM.renderToString(app);
        htmlString = htmlTemplate(htmlString, store.getState());
        res.send(htmlString);
      }).catch(function(error) {
        console.error(error);
        res.status(500).send("500 Internal error");
      });
    } else {
      res.status(404).send("404 Not Found");
    }
  });

```

Common code:

```javascript
// routes.js
const routes = [{
  path: '/',
  component: App,
  indexRoute: { onEnter: (nextState, replace) => replace('component1') },
  childRoutes: [
    { path: 'component1', component: Component1 },
    { path: 'component2', component: Component2 }
  ]
}];

// app.js
const activeStyle = {color: 'black'};

export const App = ({children}) => {
  return (<div>
        <Link to="/component1" activeStyle={activeStyle}> Component1 </Link>
        <Link to="/component2" activeStyle={activeStyle}> Component2 </Link>
        {children}
  </div>);
}
```

The difference is that on the server activeStyle looks to be always set.
On the client it works as expected.

Can somebody reproduce it?

**Comment by Vanuan** [comment]
Looks like the context is somehow propagated to unrelated components.

**Comment by jbaxleyiii** [comment]
@Vanuan I'll take a look!

**Comment by jbaxleyiii** [comment]
@Vanuan this should be fixed in #120
